cmd/pmm-dump/util.go

```diff
@@ -172,14 +184,14 @@ func getPMMServiceAgentsIds(pmmURL string, c *grafana.Client, serviceID string)
 	if statusCode != fasthttp.StatusOK {
 		return nil, fmt.Errorf("non-ok status: %d", statusCode)
 	}
-	agentsResp := new(agentsRespStruct)
-	if err = json.Unmarshal(body, agentsResp); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal response: %s", err)
+	var agentsResp agentsRespStruct
+	if err = json.Unmarshal(body, &agentsResp); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal response: %w", err)
 	}
 
 	agentsIDs := make([]string, 0)
 
-	for _, agentType := range *agentsResp {
+	for _, agentType := range agentsResp {
 		for _, agent := range agentType {
 			if agent.ServiceID != nil && *agent.ServiceID == serviceID {
```

Comment:
Hi @pooknull, you could use service_id filter to not filter on pmm-dump side. https://github.com/percona/pmm/blob/main/api/inventorypb/json/inventorypb.json#L7602
And also you could use our generated code to not re-implement it on your side https://github.com/percona/pmm/blob/main/api/inventorypb/json/client/pmm_inventory_api_client.go